homeassistant/components/vera/__init__.py

```diff
@@ -50,21 +51,23 @@
 )
 
 
-def setup(hass: HomeAssistant, base_config: dict) -> bool:
+async def async_setup(hass: HomeAssistant, base_config: dict) -> bool:
     """Set up for Vera controllers."""
-    config = base_config.get(DOMAIN, [])
+    config = base_config.get(DOMAIN)
 
     # Normalize the base url.
     config[CONF_CONTROLLER] = config.get(CONF_CONTROLLER).rstrip("/")
+    base_url = config.get(CONF_CONTROLLER)
+
+    controller, _ = veraApi.init_controller(base_url)
```

Comment:
We shouldn't create the controller here. It should be created in `async_setup_entry` and in the config flow.